### PR TITLE
Refactor MTE-2557 SettingsAppearanceTest and TestRail

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2557)

## :bulb: Description
Ensure `SettingsAppearanceTests` have corresponding TestRail tests:
https://docs.google.com/document/d/1QDyaBvyJtxrLdyQIK4b3ZM9vHuuTiYnFDkH9ZqW2eWQ/edit

I have also uncommented some steps that are now working and applicable.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: focus-ios/focus-ios-tests/XCUITest/SettingAppearanceTest.swift [hunk 1]
```diff
@@ -106,43 +107,44 @@ class SettingAppearanceTest: BaseTestCase {
...
 
+    // https://testrail.stage.mozaws.net/index.php?/cases/view/2574974
     func testOpenInSafari() {
         let safariapp = XCUIApplication(privateWithPath: nil, bundleID: "com.apple.mobilesafari")!
         loadWebPage("https://www.google.com", waitForLoadToFinish: true)
 
         waitForExistence(app.buttons["HomeView.settingsButton"])
         app.buttons["HomeView.settingsButton"].tap()
-        let safariButton = app.collectionViews.cells.element(boundBy: 5)
+        let safariButton = app.collectionViews.buttons["Open in Default Browser"]
```
Comment (clarmso) [inline focus-ios/focus-ios-tests/XCUITest/SettingAppearanceTest.swift:128]: The order of menu item has been changed. I guess this step is to "Open in Default Browser". `5` is "Find in Page".

File: focus-ios/focus-ios-tests/XCUITest/SettingAppearanceTest.swift [hunk 2]
```diff
@@ -106,43 +107,44 @@ class SettingAppearanceTest: BaseTestCase {
...
+        app.activate()
 
         // Now back to Focus
-        // waitForWebPageLoad()
-        // app.buttons["URLBar.deleteButton"].tap()
-        // waitForExistence(app.staticTexts["Your browsing history has been erased."])
+        waitForWebPageLoad()
+        waitForExistence(app.buttons["URLBar.deleteButton"])
+        app.buttons["URLBar.deleteButton"].tap()
+        waitForExistence(app.staticTexts["Browsing history cleared"])
```
Comment (clarmso) [inline focus-ios/focus-ios-tests/XCUITest/SettingAppearanceTest.swift:143]: I can now relaunch Focus and clear history.

**Comment by isabelrios** [comment]
As commented in the Jira ticket (https://mozilla-hub.atlassian.net/browse/MTE-2557) the settings tests are in different test rail test suites. And the automated tests are not only about settings Appearance because there are some related to Search Options. 
So I would rename this automated test suite as SettingsTest like the general Settings test suite in Test Rail.

I would also move the autocomple related tests in websiteAccessTest, like this one: https://github.com/mozilla-mobile/firefox-ios/blob/bf4b64d54ef7dbfb77d73e17a94b9f8f0efe3186/focus-ios/focus-ios-tests/XCUITest/WebsiteAccessTest.swift#L63 (and the other two) into this test suite because in Test Rail URL Autocomplete test suite is part of Settings.

File: focus-ios/focus-ios-tests/XCUITest/SettingAppearanceTest.swift [hunk 3]
```diff
@@ -267,4 +275,154 @@ class SettingAppearanceTest: BaseTestCase {
...
+        if desiredAutoCompleteState != topSitesState {
+            mozTap(autoCompleteSwitch)
+        }
+
+        // Navigate back to home page
+        mozTap(settingsBackButton)
+        mozTap(settingsDoneButton)
+    }
+
+    // Smoketest
```
Comment (isabelrios) [inline focus-ios/focus-ios-tests/XCUITest/SettingAppearanceTest.swift:303]: Nit: We are missing the test rail link here...

**Review by isabelrios — CHANGES_REQUESTED** [review]
Please change the Test suite name and add the missing link

File: focus-ios/focus-ios-tests/FullFunctionalTests.xctestplan [hunk 4]
```diff
@@ -30,8 +30,9 @@
         "PastenGoTest\/testPastenGo()",
         "RequestDesktopTest\/testActivityMenuRequestDesktopItem()",
         "SearchProviderTest\/testGoogleSearchProvider()",
-        "SettingAppearanceTest\/testCheckSetting()",
-        "SettingAppearanceTest\/testSafariIntegration()",
+        "SettingTest\/testCheckSetting()",
+        "SettingTest\/testSafariIntegration()",
+        "SettingTest\/testVisitWebsite()",
```
Comment (clarmso) [inline focus-ios/focus-ios-tests/FullFunctionalTests.xctestplan:35]: TestRail says `testVisitWebsite()` should be a smoke test.

File: focus-ios/focus-ios-tests/SmokeTest.xctestplan [hunk 5]
```diff
@@ -20,17 +20,20 @@
...
-        "SettingAppearanceTest\/testAddRemoveCustomDomain()",
-        "SettingAppearanceTest\/testDisableAutocomplete()",
-        "SettingAppearanceTest\/testOpenInSafari()",
-        "PageShortcutsTest\/testAddRemoveShortcut()",
+        "SettingTest\/testAddRemoveCustomDomain()",
+        "SettingTest\/testAutocompleteCustomDomain()",
+        "SettingTest\/testDisableAutocomplete()",
+        "SettingTest\/testEnableDisableAutocomplete()",
+        "SettingTest\/testOpenInSafari()",
+        "SettingTest\/testReEnableAutoComplete()",
```
Comment (clarmso) [inline focus-ios/focus-ios-tests/SmokeTest.xctestplan:36]: I ensure that these `SettingTest`s are the full functional tests according to TestRail.

**Review by isabelrios — APPROVED** [review]

**Pull request merged**